### PR TITLE
Support gateway UI

### PR DESCRIPTION
**Pull request by sinexu**

### CODE REVIEW

File: config/file.go
Comment (imangoodarzi) [inline config/file.go]: shouldn't we return err here?

File: config/operation.go
Comment (imangoodarzi) [inline config/operation.go]: 2019? :)

File: config/operation.go
Comment (imangoodarzi) [inline config/operation.go]: general idea about all error strings in this file: I don't know how and where are these errors are going to be surfaced for the users. If it is as part of the command line interface, then they look fine. But you might want to consider adding more context to these by including the actual service name in this case or the other variables inside the error message.

Obviously you are more familiar than me in this code base and I will leave this up to you to decide. :)

File: config/operation.go
Comment (imangoodarzi) [inline config/operation.go:437]: is this the only reason that parsing might fail or is the most common one?

File: config/operation.go
Comment (imangoodarzi) [inline config/operation.go]: regex compilations are usually expensive and it might worth if you take this out and define/compile it only once and use it multiple times here.

File: config/operation.go
Comment (imangoodarzi) [inline config/operation.go]: ditto here too.

File: server/template.go
Comment (imangoodarzi) [inline server/template.go]: ditto

File: server/ui.go
Comment (imangoodarzi) [inline server/ui.go:62]: what is this id/secret used for? do we need to further secure them in a key vault or something?

**Review by imangoodarzi — COMMENTED** [review]
Wonderful job Leesa. Big PR with a lot of code. I did one pass and left some minor comments.

File: server/ui.go
Comment (sinexu) [reply server/ui.go:62]: according to the security review, it seems fine, i will keep an eye on this

File: config/operation.go
Comment (sinexu) [reply config/operation.go]: this file is used in both terminal file and ui file, well... logically UI should never show a page with these error messages displayed (mostly for debug use); for terminal, yes it would show the users. I will add some information. Thanks!

File: config/operation.go
Comment (sinexu) [reply config/operation.go:437]: the only hard restriction here is it has to be "https" for all the urls, we don't have other information

File: config/operation.go
Comment (sinexu) [reply config/operation.go]: thanks!
i didnt think about this

File: server/server.go
Comment (imangoodarzi) [inline server/server.go]: can this be named "handlers" or something more relevant instead of "H"?

File: server/server.go
Comment (imangoodarzi) [inline server/server.go:110]: ditto here and the functions below as well.

**Review by imangoodarzi — APPROVED** [review]

**Pull request merged**
